packages/core/src/components/AvatarGroup/AvatarGroup.tsx

```diff
@@ -0,0 +1,155 @@
+import { Children, cloneElement, CSSProperties, forwardRef } from "react";
+
+import { css } from "@emotion/css";
+
+import { HvSize, theme } from "@hitachivantara/uikit-styles";
+
+import { useDefaultProps } from "@core/hooks/useDefaultProps";
+
+import { HvBaseProps } from "@core/types/generic";
+
+import { ExtractNames } from "@core/utils/classes";
+
+import { staticClasses, useClasses } from "./AvatarGroup.styles";
+
+import { HvAvatar } from "../Avatar/Avatar";
+
+export { staticClasses as avatarGroupClasses };
+
+export type HvAvatarGroupClasses = ExtractNames<typeof useClasses>;
+
+export interface HvAvatarGroupProps extends HvBaseProps {
+  /** Inline styles to be applied to the root element. */
+  style?: CSSProperties;
+  /** A Jss Object used to override or extend the styles applied to the component. */
+  classes?: HvAvatarGroupClasses;
+  /** The avatar size. */
+  size?: HvSize;
+  /** The spacing between avatars. */
+  spacing?: "compact" | "loose";
+  /** The direction of the group. */
+  direction?: "row" | "column";
+  /** Whether the avatars display behind the previous avatar or on top. */
+  toBack?: boolean;
+  /**
+   * The maximum number of visible avatars. If there are more avatars then the value of this property, an added avatar will
+   * be added to the end of the list, indicating the number of hidden avatars.
+   */
+  maxVisible?: number;
+  /**
+   * What to show as an overflow representation.
+   * If `undefined` a default `HvAvatar` will be displayed along with a HvTooltip with the count of overflowing items.
+   * */
+  overflowComponent?: (overflowCount: number) => React.ReactNode;
+}
+
+const getSpacingValue = (
+  spacing: HvAvatarGroupProps["spacing"],
+  size: HvAvatarGroupProps["size"]
+) => {
+  switch (size) {
+    case "xs":
+      return spacing === "compact" ? 24 : 16;
+    case "sm":
+      return spacing === "compact" ? 30 : 18;
+    case "md":
+      return spacing === "compact" ? 36 : 20;
+    case "lg":
+      return spacing === "compact" ? 44 : 24;
+    case "xl":
+      return spacing === "compact" ? 72 : 34;
+    default:
+      return spacing === "compact" ? 30 : 18;
+  }
+};
+
+/**
+ * The AvatarGroup component is used to group multiple avatars.
+ */
+export const HvAvatarGroup = forwardRef<any, HvAvatarGroupProps>(
+  (props, ref) => {
```

Comment:
We could use a less generic type than `any` for the `ref`, like `HTMLDivElement`